src/Razor/src/Microsoft.VisualStudio.LiveShare.Razor/Guest/ProjectSnapshotSynchronizationService.cs

```diff
@@ -16,38 +18,24 @@ internal class ProjectSnapshotSynchronizationService : ICollaborationService, IA
     private readonly JoinableTaskFactory _joinableTaskFactory;
     private readonly CollaborationSession _sessionContext;
     private readonly IProjectSnapshotManagerProxy _hostProjectManagerProxy;
-    private readonly ProjectSnapshotManagerBase _projectSnapshotManager;
+    private readonly IProjectSnapshotManagerAccessor _projectManagerAccessor;
+    private readonly IErrorReporter _errorReporter;
+    private readonly ProjectSnapshotManagerDispatcher _dispatcher;
 
     public ProjectSnapshotSynchronizationService(
         JoinableTaskFactory joinableTaskFactory,
         CollaborationSession sessionContext,
         IProjectSnapshotManagerProxy hostProjectManagerProxy,
-        ProjectSnapshotManagerBase projectSnapshotManager)
+        IProjectSnapshotManagerAccessor projectManagerAccessor,
+        IErrorReporter errorReporter,
+        ProjectSnapshotManagerDispatcher dispatcher)
     {
-        if (joinableTaskFactory is null)
```

Comment:
nit: is the null check removal intentional?

Reply:
Yeah. IMO, a lot of the argument null checks are kind of gratuitous. This isn't a public API and nullability is enabled for this type, so I don't see a lot of value in throwing on null here. Happy to change if you feel strongly though.

Reply:
I agree, but wanted to make sure it wasn't an accidental change :)

Reply:
Thanks for checking, and apologies if the extra clean up made this harder to review in any way.

Reply:
Remove them all I say! (if nullability is enabled)

Reply:
Not at all. It empowers me to do the same in future PRs ;)

Reply:
As far as I know, MS.Ext.DI will also fail at composition creation time if there are services missing, and since we have no public API, I see no reason for any argument checking in any of our annotated code.

Reply:
Yes, I believe you're right about that.